core/src/main/java/com/radixdlt/rev2/REv2StateComputer.java

```diff
@@ -225,59 +225,67 @@ public StateComputerLedger.StateComputerResult prepare(
 
   @Override
   public void commit(
-      CommittedTransactionsWithProof txnsAndProof, VertexStoreState vertexStoreState) {
-    var proofBytes = serialization.toDson(txnsAndProof.getProof(), DsonOutput.Output.ALL);
-    final Option<byte[]> vertexStoreBytes;
-    if (vertexStoreState != null) {
-      vertexStoreBytes =
-          Option.some(serialization.toDson(vertexStoreState.toSerialized(), DsonOutput.Output.ALL));
+      CommittedTransactionsWithProof txnsAndProof, VertexStoreState preCommitVertexStoreState) {
+    final var proof = txnsAndProof.getProof();
+
+    final var outputBuilder = ImmutableClassToInstanceMap.builder();
+
+    final VertexStoreState postCommitVertexStoreState;
+    if (proof.getNextEpoch().isPresent()) {
+      final var nextEpoch = proof.getNextEpoch().get();
+      final var initialEpochVertex =
+          Vertex.createInitialEpochVertex(proof.getHeader()).withId(hasher);
+      final var nextLedgerHeader =
+          LedgerHeader.create(
+              nextEpoch.getEpoch(),
+              Round.genesis(),
+              proof.getAccumulatorState(),
+              proof.consensusParentRoundTimestamp(),
+              proof.proposerTimestamp());
+      final var initialEpochQC =
+          QuorumCertificate.createInitialEpochQC(initialEpochVertex, nextLedgerHeader);
+
+      // Post commit vertex store is a fresh one
+      postCommitVertexStoreState =
+          VertexStoreState.create(
+              HighQC.ofInitialEpochQc(initialEpochQC), initialEpochVertex, hasher);
```

Comment:
I appreciate this code is just mostly a move, but it might be clearer if we can encapsulate this logic into a new function, eg `VertexStoreState.createNewForNextEpoch(nextEpoch)` ?

Reply:
:+1: done